File: 38/solution.py
#!/usr/bin/env python
#
# Project Euler 38

def isPandigital(s):
    return sorted(s) == ['1','2','3','4','5','6','7','8','9']

def generate(i):
    j, result = 1, ""
    while len(result) < 9:
        result += str(i*j)
        j += 1
    return result

print max(filter(isPandigital, map(generate, xrange(1,1000000))))

File: 38/solution.clj
;;; Project Euler 38

(defn pandigital-9? [n]
  "A pandigital number contains exactly one each of the digits 1-9"
  (= '(\1 \2 \3 \4 \5 \6 \7 \8 \9)
     (sort (str n))))

(defn generate [i]
  (new Long
       (loop [j 1 current ""]
         (if (< (.length current) 9)
           (recur (inc j) (str current (* i j)))
           current))))

(defn solution [n]
  "Largest pandigital number that can be formed by generate"
  (apply max
         (filter pandigital-9?
                 (map generate (range n)))))

(time (solution 100000))
